### DISCUSSION TITLE
Multiple MorphToMany relationships

### DISCUSSION THREAD

**Discussion by flashmediasolutions** [staudenmeir/eloquent-has-many-deep#197] [Q&A] [answered]
I am having a difficult time putting together the right format with this relationship.

I want to get all the `Examples` for a `Store` and vice versa.
It gets complicated in 2 parts.

1. They are going through an intermediary. (`Store` -> `Source A` -> `Example`)
2. There are 3 possible models the relationship could exist with `Source A` or `Source B` or `Source C`

Any help, or any advice on a better way would be greatly appreciated!

Thanks

![Screenshot 2023-06-15 at 2 02 22 PM](https://github.com/staudenmeir/eloquent-has-many-deep/assets/12239388/25b91281-ef99-4ab0-b6f9-7bf36f613dc7)

**Comment by staudenmeir** [comment, marked as answer]
Hi @flashmediasolutions,
Combine this package with one of my other ones:
https://github.com/staudenmeir/laravel-merged-relations

1. Define the relationships:

```php
class Store extends Model
{
    use \Staudenmeir\EloquentHasManyDeep\HasRelationships;
    use \Staudenmeir\LaravelMergedRelations\Eloquent\HasMergedRelationships;

    public function examples()
    {
        return $this->mergedRelationWithModel(Example::class, 'all_examples');
    }

    public function examplesA()
    {
        return $this->hasManyDeepFromRelations($this->sourcesA(), (new SourceA())->examples());
    }

    public function examplesB()
    {
        return $this->hasManyDeepFromRelations($this->sourcesB(), (new SourceB())->examples());
    }

    public function examplesC()
    {
        return $this->hasManyDeepFromRelations($this->sourcesC(), (new SourceC())->examples());
    }

    public function sourcesA()
    {
        return $this->morphedByMany(SourceA::class, 'storeable');
    }

    public function sourcesB()
    {
        return $this->morphedByMany(SourceB::class, 'storeable');
    }

    public function sourcesC()
    {
        return $this->morphedByMany(SourceC::class, 'storeable');
    }
}
```

2. Create the merge view in a migration:

```php
use Staudenmeir\LaravelMergedRelations\Facades\Schema;

Schema:: createOrReplaceMergeView(
    'all_examples',
    [(new Store())->examplesA(), (new Store())->examplesB(), (new Store())->examplesC()]
);
```

3. Use the merged relationship:

```php
$allExamples = Store::find($id)->examples;
```

If that works for you, we'll take a look at the reverse relationship.

**Reply by flashmediasolutions** [reply]
That worked great! There is no way I figure that out on my own.

Thank you very much for the help!
